src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -25,38 +25,48 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceScopeFac
                 throw new ArgumentNullException(nameof(scopeFactory));
             }
 
-            _scopeFactory = scopeFactory;
             _next = next;
+            _scopeFactory = scopeFactory;
         }
 
         public async Task Invoke(HttpContext httpContext)
         {
             if (httpContext == null)
```

Comment:
This can't really happen either. I wonder if we should just Debug.Assert here as well. We don't check for a null context in our middleware in general...